pyflagser/tests/test_flagser.py

```diff
@@ -182,22 +188,17 @@ def are_matrix_equal(m1, m2):
 def test_filtrations(flag_file):
     """
     Testing all filtrations available for dataset d5.flag
-    vertex_degree filtrations was disable because it produces a segmentation
-    fault.
     """
     if os.path.split(flag_file)[1] == 'd5.flag':
         flag_matrix = loadflag(flag_file)
         for filtration in implemented_filtrations:
```

Comment:
The loop should instead be implemented using `pytest.mark.parametrize`. I'll make the changes if OK with you?

Reply:
I looked a bit myself (I'm curious), the only way I found to parametrize is as follow:

```python
@pytest.mark.parametrize("filtration",  implemented_filtrations)
def test_filtrations(flag_file, filtration):
    """
    Testing all filtrations available for dataset d5.flag
    """
    if os.path.split(flag_file)[1] == 'd5.flag':
        flag_matrix = loadflag(flag_file)
        res = flagser(flag_matrix, max_dimension=1, directed=False,
                      filtration=filtration)
        for filt, tests in filtrations_results.items():
            if filtration == filt:
                tmp = np.array(res['dgms']).tolist()
                tmp2 = np.array(tests['dgms']).tolist()
                assert are_matrix_equal(tmp, tmp2), \
                    "diagrams {} \n and {} \n are not equal"\
                    .format(tmp, tmp2)
    else:
        pytest.skip(
            'file {} not currently implemented for firltration'.format(
                flag_file))
```

But then I get the following output when running `pytest`:

```bash
collected 187 items

pyflagser/tests/test_flagser.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [  0%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssss...........ssssssssssssssssssssssssssss [  0%]
sssssssssssssssssssssssssss
========================= 11 passed, 176 skipped, 22 warnings in 0.42 seconds =========================
```

It's a lot of noise, just because it performs a product between the number of input files and the number of filtrations.

Let me know if you have a better solution

Reply:
I think all this noise is generated by the `else` statement. Why not just remove it completely?

Reply:
If you remove the `else` statement I obtain:

```bash
pyflagser/tests/test_flagser.py ................................................................ [  0%]
................................................................................................ [  0%]
...........................
=============================== 187 passed, 22 warnings in 0.60 seconds ===============================
```

The problem is at the generation of the test, I can have a look to better specify test.

Reply:
@MonkeyBreaker I pushed some changes in 3ce9fc2 which should allow us to have parametrizations while not seeing noise. Let me know what you think!

Reply:
Nice, it works like a charm on my computer !
For me, we can merge this PR